src/schemathesis/cli/__init__.py

```diff
@@ -168,11 +171,12 @@ def run(  # pylint: disable=too-many-arguments
 
     with abort_on_network_errors():
         if pathlib.Path(schema).is_file():
-            prepared_runner = runner.prepare(
-                schema, checks=selected_checks, workers_num=workers_num, loader=from_path, **options
-            )
+            options.update({"checks": selected_checks, "workers_num": workers_num, "loader": from_path})
         else:
-            prepared_runner = runner.prepare(schema, checks=selected_checks, workers_num=workers_num, **options)
+            if app is not None:
+                options["loader"] = from_wsgi
+            options.update({"checks": selected_checks, "workers_num": workers_num})
```

Comment:
Seems like checks and workers_num are present in both branches - could be moved before the condition